teos/chain_monitor.py

```diff
@@ -1,22 +1,43 @@
+from enum import Enum
+from queue import Queue, Empty
 import zmq
 import binascii
 from threading import Thread, Event, Condition
 
 from common.logger import get_logger
 
 
+class ChainMonitorStatus(Enum):
+    IDLE = 0
+    LISTENING = 1
+    ACTIVE = 2
+    TERMINATED = 3
+
+
 class ChainMonitor:
     """
     The :class:`ChainMonitor` is in charge of monitoring the blockchain (via ``bitcoind``) to detect new blocks on top
-    of the best chain. If a new best block is spotted, the chain monitor will notify the
-    :obj:`Watcher <teos.watcher.Watcher>` and the :obj:`Responder <teos.responder.Responder>` using ``Queues``.
+    of the best chain. If a new best block is spotted, the chain monitor will notify the given ``Queues``.
 
     The :class:`ChainMonitor` monitors the chain using two methods: ``zmq`` and ``polling``. Blocks are only notified
     once per queue and the notification is triggered by the method that detects the block faster.
 
+    The :class:`ChainMonitor` lifecycle goes through 4 states: idle, listening, active and terminated.
```

Comment:
Should be done, although I'm still not 100% sure when we're using single and when dobule quotes. Is there a place where I can read that?

Reply:
I think this may be the best place https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google

The rule of thumb is: double quote variable names you'd like to highlight, single quotes only work within class / object references (e.g: :obj:`Name <path>`)

Reply:
There are still some btw:

L27: ChainMonitorStatus.IDLE
L29: ChainMonitorStatus.LISTENING
L53-54:  IDLE, LISTENING, ACTIVE and TERMINATED

Reply:
Fixed in 3dabf51.